volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -1102,10 +1116,24 @@ def d_ancestor(self, ancestor_dentry):
             not current_dentry.is_root()
             and current_dentry.vol.offset not in dentry_seen
         ):
-            if current_dentry.d_parent == ancestor_dentry.vol.offset:
+            try:
```

Comment:
Again, feels like a single try/except could wrap all the statements here because the middle statements don't matter if it's still going to return None on a failure?